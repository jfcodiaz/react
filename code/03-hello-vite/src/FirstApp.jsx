import PropTypes from 'prop-types';

// Usando destructuring con valores por defecto (ES6) - recomendado para React 19
export default function FirstApp({
    title = 'No hay titulo', 
    subTitle = 0, 
    name = "Pako Díaz"
}) {
    return (
        <>
            <h1>{ title }</h1>
            <p>{ subTitle }</p>
            <p>Autor: { name }</p>
        </>
    );
}

// PropTypes para validación en desarrollo
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number.isRequired,
    name: PropTypes.string
};

// Nota: defaultProps está deprecado en React 19+
// Se recomienda usar valores por defecto en la destructuring de parámetros