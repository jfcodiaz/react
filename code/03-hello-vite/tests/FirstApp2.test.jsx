import { render, screen} from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe('Prueba en <FirstApp />', () => {
    const title = "Hola, soy Goku";
    const subTitle = "Soy un guerrero Saiyajin";

    test('debe de hacer match con el snapshot', () => {
        //screen.debug();
        const { container } = render(<FirstApp title={title}/>);
        //screen.debug();
        expect(container).toMatchSnapshot();  
        //creen.debug(); // Para ver el DOM renderizado 
    });

    test('debe de mostrar el mensaje "Hola, soy Goku"', () => {
        render(<FirstApp title={title}/>);
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('debe de mostrar el título en un h1', () => {
        render(<FirstApp title={title}/>);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('debe de mostrar el subtítulo enviado por props', () => {
        render(<FirstApp title={title} subTitle={subTitle}/>);
        expect(screen.getByText(subTitle)).toBeTruthy();
    });
});