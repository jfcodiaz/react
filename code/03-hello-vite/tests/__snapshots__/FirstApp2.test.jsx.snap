// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Prueba en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    0
  </p>
  <p>
    Autor: 
    Pako Díaz
  </p>
</div>
`;

exports[`Prueba en <FirstApp /> debe mostrar el título correctamente 1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    0
  </p>
  <p>
    Autor: 
    Pako Díaz
  </p>
</div>
`;
